File: 349-intersection/Solution.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution349 {
    public int[] P0349_intersection(int[] nums1, int[] nums2) {
        Set<Integer> nums1Set = new HashSet<>();
        Set<Integer> intersect = new HashSet<>();
        for (int n : nums1){
            nums1Set.add(n);
        }
        for (int n : nums2){
            if (nums1Set.contains(n)){
                intersect.add(n);
                nums1Set.remove(n);
            }
        }

        int[] result = new int[intersect.size()];//??
        int counter = 0;
        for (Integer i : intersect){
            result[counter++] = i;
        }
        return result;
    }
    static void main(String[] args) {
        Solution349 sol = new Solution349();
        int[] numbs1 = {4,9,5};
        int[] numbs2 = {9,4,9,8,4};
        System.out.println(Arrays.toString(sol.P0349_intersection(numbs1, numbs2)));
    }
}

File: 349-intersection/solution.py
from typing import List
class Solution1122:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1_set = set()        
        intersect = []
        for i in nums1:
            nums1_set.add(i)
            #print(nums1_set)

        for i in nums2:
            if i in nums1_set:
                intersect.append(i)
                nums1_set.discard(i)
        return intersect

nums1 = [4,9,5]
nums2 = [9,4,9,8,4]
print(Solution1122().intersection(nums1, nums2))

"""
输入：nums1 = [4,9,5], nums2 = [9,4,9,8,4]
输出：[9,4]
解释：[4,9] 也是可通过的
"""